lib/bolt/plan_future.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'fiber'
+
+module Bolt
+  class PlanFuture
+    attr_reader :fiber, :id
+    attr_accessor :value
+
+    def initialize(fiber, id, name = nil)
+      @fiber = fiber
+      @id    = id
+      @name  = name
+      @value = nil
+    end
+
+    def name
+      @name || @id
+    end
+
+    def to_s
+      "Future '#{name}'"
+    end
+
+    def alive?
+      fiber.alive?
+    end
+
+    def raise(exception)
+      fiber.raise(exception)
```

Comment:
I can't remember, but did we need to maintain compatibility with Ruby 2.5? `Fiber.raise` is private before 2.7. If not, please ignore. :)

Reply:
We do in general need to maintain compatibility with Ruby 2.5, but the whole feature won't be supported in other projects for the foreseeable future - I'll leave a note this will only work in Ruby 2.7, but I don't expect it to be an issue for a while. There's also no risk of this getting called by running the main plan because this is only ever used to raise a Timeout error for `wait`, which is used by the `wait()` and `parallelize()` plan functions.